src/sqs/backfill-error-handler.test.ts

```diff
@@ -116,4 +118,30 @@ describe("backfillErrorHandler", () => {
 		expect(sendMessageMock.mock.calls[0][1]).toEqual(0);
 		expect((await RepoSyncState.findByPk(repoSyncState!.id)).commitStatus).toEqual("failed");
 	});
+
+	it("marks task as failed and reschedules message on permission error", async () => {
+		const { subscription, repoSyncState } = await new DatabaseStateCreator().withActiveRepoSyncState().create();
+
+		const context = createContext(5, true);
+		context.payload = {
+			jiraHost,
+			installationId: subscription.gitHubInstallationId
+		};
+
+		const task = _.cloneDeep(TASK);
+		task.repositoryId = repoSyncState?.repoId || -1;
+
+		const sendMessageMock = jest.fn();
+		const result = await backfillErrorHandler(sendMessageMock)(new TaskError(task, new InvalidPermissionsError({ } as unknown as AxiosError)), context);
+
+		expect(result).toEqual({
+			isFailure: false
+		});
+		expect(sendMessageMock.mock.calls[0][0]).toEqual(
+			{ installationId: DatabaseStateCreator.GITHUB_INSTALLATION_ID, jiraHost }
+		);
+		expect(sendMessageMock.mock.calls[0][1]).toEqual(0);
```

Comment:
So we don't need exponential for permission retry? Even for jira?

Reply:
Oh wait, we are not retrying are we?

Reply:
That's for github only, InvalidPermissionError is a github client error. I'll rename the exception to make it more clear in my next PR

Reply:
For the permissions errors from GitHub we are not retrying, correct